### ISSUE TITLE
[master]: (spurious?) failures of tests in tsdb/engine/tsm1 when compiled with go master

### ISSUE THREAD

**Issue by jonseymour** [influxdata/influxdb#5854]
The current docker build images used by the influxdata CI appear to compile with go master. Some tests in tsdb/engine/tsm1 fail when compiled with that go version (1d5001af), but not with an earlier version
like 1.6 (7bc40ffb)

A PR which demonstrates that the build failure disappears when using earlier versions of go will be submitted.

The test break is:

```
=== RUN   Test_IntegerEncoder_Quick
--- FAIL: Test_IntegerEncoder_Quick (0.00s)
	int_test.go:443: mismatch:
		
		exp=[]
		
		got=[]
		
=== RUN   Test_StringEncoder_NoValues
--- PASS: Test_StringEncoder_NoValues (0.00s)
=== RUN   Test_StringEncoder_Single
--- PASS: Test_StringEncoder_Single (0.00s)
=== RUN   Test_StringEncoder_Multi_Compressed
--- PASS: Test_StringEncoder_Multi_Compressed (0.00s)
=== RUN   Test_StringEncoder_Quick
--- FAIL: Test_StringEncoder_Quick (0.00s)
	string_test.go:118: mismatch:
		
		exp=[]
		
		got=[]

...

=== RUN   Test_BooleanEncoder_Quick
--- FAIL: Test_BooleanEncoder_Quick (0.00s)
	bool_test.go:100: mismatch:
		
		exp=[]
		
		got=[]
				
```

**Comment by jonseymour** [comment]
Per discussion in #general on slack.gopher.com, the go change that is responsible for the changed behaviour is probably:

https://github.com/golang/go/commit/a77182f47f0e55789f06c13accd694df60d85f07

**Comment by jonseymour** [comment]
So, in summary, it seems that the behaviour of testing.quick.Check  in go tip has changed in way that is now exercising a code path that is now causing a test failure.

**Comment by jonseymour** [comment]
My proposed fix for the issue checks for this case in the Check handler and alters the expected behaviour to the current actual behaviour for this case. If this is not actually the expected behaviour, then a separate issue needs to be raised to fix the expected behaviour. 

This fix probably needs to be accepted as is, so that every other build doesn't fail.

**Comment by jonseymour** [comment]
Closed because #5855 has been merged,
